### ISSUE TITLE
Why cross-entropy doesn't have logarit ?

### ISSUE THREAD

**Issue by vinhdc10998** [kanamekojima/rnnimp#2]
Dear Kaname,

I wonder why your loss function doesn't have logarit when it is cross-entropy loss ?
![image](https://user-images.githubusercontent.com/55987491/111503392-4ac2b900-8779-11eb-8f3e-fe61a11ddc93.png)

Thank you.

**Comment by vinhdc10998** [comment]
Hi Kaname,

I'm trying to implement your paper with my dataset, so I need to train my data. I wonder how to build region_n.legend.gz in your model such as a1_freq and array_marker_flag ( because I saw some position of SNPs in file test  chr22.legend is 1 but it is 0 in file chr22_1.legend). Addtionally, can you please share me how many input in your train model ?

Thank you.

**Comment by kanamekojima** [comment]
Hi, Dương,

Thank you for the comments.

The logarithm is actually required for the loss function as in the usual cross entropy loss (I forgot to write the logarithm in the loss function by mistake).

To create the legend file, we first choose some specific SNP array (e.g., Omni2.5 or OmniExpress) and haplotype reference panel (e.g., 1KGP data, HRC data, or inhouse haplotype panel data), and set 1 for array_marker_flag if SNPs in the haplotype reference panel are designed in the SNP array and 0 otherwise. For a1_freq column, the frequency of allele 1 (a1) in the haplotype reference panel is given.

Each region where the train model is built is comprised of the center region, which contains both observed and unobserved variants, the left flanking region, and the right flaking region. Our implementation extends the center region until the number of unobserved variants reaches 1,000 or the number of observed variants reaches 100. The left and right flaking regions contain only the observed variants, i.e, variants to be imputed are located only in the center region. Our implementation extends the left flanking region until the sum of numbers of observed variants in the left flanking region and the center region reaches 200. The right flanking region is also extended in a similar manner. Thus, the number of inputs (observed variants) of the model for each region is mostly around 300, but can be less and at maximum 400.

RNN-IMP is based on a bidirectional RNN, and the forward RNN is built in the left flanking region and the center region, while the backward RNN is built in the right flaking region and the center region. In order to limit and balance the memory usage of both forward and backward RNNs, our implementation limits the number of observed variants in the above manner.

I hope those are sufficiently answering to your comments.
